### ISSUE TITLE
Pagination: Consider keeping current item a focusable button, use aria-current to specify current item

### ISSUE THREAD

**Issue by hbuchel** [aws-amplify/amplify-ui#1982]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

N/A

### Please describe your bug.

The current pagination component switches the "active" page to a span. This makes it a little awkward when navigating via screenreader (when tabbing through focusable items) because it won't ever read out the visually hidden text for that element. You could guess that the "missing" element is the current, and verify by switching to how you're navigating the page, but it seems like an improvement to just have it be a focusable button like the others and add some additional attributes.

https://user-images.githubusercontent.com/376920/170329437-9f2806b2-0e1f-4d8f-9ca7-eaadae314ea3.mp4

### What's the expected behaviour?

You could expect the current item button to be focusable, and for a screenreader to announce if the item is current or not. In that case it might look generally like:

```
<button aria-current="true"><span class="amplify-visually-hidden">Current page</span> 1<button>
```

We can check out some other similar examples:
https://designsystem.digital.gov/components/pagination/
https://mui.com/material-ui/react-pagination/
https://design-system.w3.org/components/pagination.html

### Help us reproduce the bug!

Current pagination examples https://ui.docs.amplify.aws/react/components/pagination

### Code Snippet

```typescript
// Put your code below this line.

```
